unity-renderer/Assets/Rendering/ProceduralSkybox/TimeReporter/TimeReporter.cs

```diff
@@ -0,0 +1,48 @@
+using System;
+using DCL;
+using DCL.Interface;
+
+public class TimeReporter : IDisposable
+{
+    private float timeNormalizationFactor;
+    private float cycleTime;
+    private bool wasPaused = true;
+
+    internal event Action<float, bool> OnReport;
+
+    public TimeReporter()
+    {
+        OnReport += DoReport;
+    }
+
+    public void Dispose()
+    {
+        OnReport -= DoReport;
+    }
+
+    public void Configure(float normalizationFactor, float cycle)
+    {
+        timeNormalizationFactor = normalizationFactor;
+        cycleTime = cycle;
+    }
+
+    public void ReportTime(float time)
+    {
+        bool isPaused = !DataStore.i.skyboxConfig.useDynamicSkybox.Get();
+
+        // NOTE: if not paused and pause state didn't change there is no need to report
+        // current time since it's being calculated on kernel side
+        if (!isPaused && !wasPaused)
+        {
+            return;
+        }
```

Comment:
Cannot we just report when the time has manually changed or the skybox has been paused/resumed?

With a timestamp and an offset we should be able to calculate the rest.

Reply:
Okay, I think we can forget the above, unless I'm mistaken, it's already coded that way.

Reply:
yeap, it's already coded that way :)